merk/src/proofs/query/mod.rs

```diff
@@ -975,60 +1047,118 @@ pub fn verify_query(
                 // get next item in query
                 let query_item = *item;
                 let (lower_bound, start_non_inclusive) = query_item.lower_bound();
+                let (upper_bound, end_inclusive) = query_item.upper_bound();
 
-                // we have not reached next queried part of tree
-                if *query_item > key.as_slice() {
-                    // continue to next push
-                    break;
-                } else if start_non_inclusive
-                    && lower_bound.is_some()
-                    && lower_bound.unwrap() == key.as_slice()
-                {
-                    // we intersect with the query_item but at the start which is non inclusive
-                    // continue to the next push
-                    break;
+                if left_to_right {
+                    // we have not reached next queried part of tree
+                    if *query_item > key.as_slice() {
+                        // continue to next push
+                        break;
+                    } else if start_non_inclusive
+                        && lower_bound.is_some()
+                        && lower_bound.unwrap() == key.as_slice()
```

Comment:
I see you used `Some(key.as_slice()) == x` somewhere below, will it work here also instead of `unwrap` / `is_some` combo?